temoa/_internal/table_writer.py

```
@@ -96,425 +210,385 @@ def write_results(
-        self.write_objective(model, iteration=iteration)
-        # analyze the emissions to get the costs and flows
-        e_costs, e_flows = poll_emissions(model=model)
-        self.emission_register = e_flows
-        self.write_emissions(iteration=iteration)
-        self.con.commit()
-        self.con.execute('VACUUM')
+        try:
+            if not self.tech_sectors:
+                self._set_tech_sectors()
+            self.write_objective(model, iteration=iteration)
+            e_costs, e_flows = poll_emissions(model=model)
```

Ruff (0.14.8)

Unpacked variable `e_costs` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)